qtconsole/manager.py

```diff
@@ -41,13 +45,16 @@ def start_restarter(self):
                     parent=self,
                     log=self.log,
                 )
-                self._restarter.add_callback(self._handle_kernel_restarted)
+            self._is_restarting = True
             self._restarter.start()
 
     def stop_restarter(self):
         if self.autorestart:
             if self._restarter is not None:
                 self._restarter.stop()
 
-    def _handle_kernel_restarted(self):
-        self.kernel_restarted.emit()
+    def post_start_kernel(self, **kw):
```

Comment:
When is this new method called?

Reply:
After the kernel is started:
 https://github.com/jupyter/jupyter_client/blob/842fa46e9507d1d85aacd224d672cb624188c15f/jupyter_client/manager.py#L312

Reply:
For ref the callback is called here: https://github.com/jupyter/jupyter_client/blob/44980c13680f4e4226cf25f199ce4e4bb6e11296/jupyter_client/restarter.py#L112
BEFORE the kernel is started